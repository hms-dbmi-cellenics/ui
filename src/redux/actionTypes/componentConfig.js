/**
 * Sets plot configs and data.
 */
const COMPONENT_CONFIG = 'componentConfig';
const LOAD_CONFIG = `${COMPONENT_CONFIG}/load`;
const UPDATE_CONFIG = `${COMPONENT_CONFIG}/update`;
const RESET_CONFIG = `${COMPONENT_CONFIG}/reset`;
const SAVE_CONFIG = `${COMPONENT_CONFIG}/save`;
const PLOT_DATA_LOADED = `${COMPONENT_CONFIG}/plotDataLoaded`;
const PLOT_DATA_LOADING = `${COMPONENT_CONFIG}/plotDataLoading`;
const PLOT_DATA_ERROR = `${COMPONENT_CONFIG}/plotDataError`;

export {
  LOAD_CONFIG, UPDATE_CONFIG, RESET_CONFIG, SAVE_CONFIG,
  PLOT_DATA_LOADED, PLOT_DATA_LOADING, PLOT_DATA_ERROR,
};
