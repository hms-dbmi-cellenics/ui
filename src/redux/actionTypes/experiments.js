const EXPERIMENTS = 'experiments';

/**
 * Create experiments
 */
const EXPERIMENTS_CREATED = `${EXPERIMENTS}/created`;

/**
 * Create experiments
 */
const EXPERIMENTS_LOADING = `${EXPERIMENTS}/loading`;

/**
 * Create experiments
 */
const EXPERIMENTS_LOADED = `${EXPERIMENTS}/loaded`;

/**
 * Experiments updated
 */
const EXPERIMENTS_UPDATED = `${EXPERIMENTS}/updated`;

/**
 * Delete experiments
 */
const EXPERIMENTS_DELETED = `${EXPERIMENTS}/deleted`;

/**
 * Saving experiment
 */
const EXPERIMENTS_SAVING = `${EXPERIMENTS}/saving`;

/**
 * Update experiments
 */
const EXPERIMENTS_ERROR = `${EXPERIMENTS}/error`;

const EXPERIMENTS_SWITCH = `${EXPERIMENTS}/switch`;

export {
  EXPERIMENTS_CREATED,
  EXPERIMENTS_UPDATED,
  EXPERIMENTS_ERROR,
  EXPERIMENTS_LOADING,
  EXPERIMENTS_LOADED,
  EXPERIMENTS_DELETED,
  EXPERIMENTS_SAVING,
  EXPERIMENTS_SWITCH,
};
