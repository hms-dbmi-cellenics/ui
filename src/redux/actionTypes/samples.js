const SAMPLES = 'samples';

/**
 * Save sample.
 */
const SAMPLES_CREATE = `${SAMPLES}/create`;

/**
 * Update sample.
 */
const SAMPLES_UPDATE = `${SAMPLES}/update`;

/**
 * Delete sample.
 */
const SAMPLES_DELETE_API_V2 = `${SAMPLES}/deleteApiV2`;

/**
 * Delete sample following apiv1 version, remove when doing the final step of cleaning up v1 code.
 */
const SAMPLES_DELETE_API_V1 = `${SAMPLES}/deleteApiV1`;

/**
 * Update files in sample.
 */
const SAMPLES_FILE_UPDATE = `${SAMPLES}/fileUpdate`;

/**
 * Load sample.
 */
const SAMPLES_LOADED = `${SAMPLES}/loaded`;

/**
 * Saving samples.
 */
const SAMPLES_SAVING = `${SAMPLES}/saving`;

/**
 * Samples saved successfully.
 */
const SAMPLES_SAVED = `${SAMPLES}/saved`;

/**
 * Error loading/saving sample.
 */
const SAMPLES_ERROR = `${SAMPLES}/error`;

/**
 * Delete metada from sample.
 */
const SAMPLES_METADATA_DELETE = `${SAMPLES}/metadataDelete`;

const SAMPLES_LOADING = `${SAMPLES}/loading`;
export {
  SAMPLES_CREATE,
  SAMPLES_UPDATE,
  SAMPLES_DELETE_API_V2,
  SAMPLES_DELETE_API_V1,
  SAMPLES_FILE_UPDATE,
  SAMPLES_LOADED,
  SAMPLES_SAVING,
  SAMPLES_ERROR,
  SAMPLES_SAVED,
  SAMPLES_METADATA_DELETE,
  SAMPLES_LOADING,
};
