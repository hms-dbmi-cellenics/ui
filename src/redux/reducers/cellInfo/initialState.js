const initialState = {
  component: null,
  cellId: null,
  geneName: null,
  trackCluster: null,
  focus: {
    store: 'cellSets',
    key: 'louvain',
  },
  groupedTrack: 'louvain',
  selectedTracks: ['louvain'],
};

export default initialState;
