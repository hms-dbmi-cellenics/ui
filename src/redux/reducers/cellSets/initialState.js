const initialState = {
  properties: {},
  hierarchy: [],
  selected: {},
  initialLoadPending: true,
  loading: false,
  error: false,
  updatingClustering: false,
  hidden: new Set(),
};

export default initialState;
