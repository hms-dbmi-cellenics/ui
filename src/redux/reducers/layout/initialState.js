const initialLayoutSpatial = {
  windows: {
    direction: 'row',
    first: {
      first: {
        first: 'UMAP', second: 'Spatial', direction: 'row', splitPercentage: 62,
      },
      second: {
        first: 'Heatmap',
        second: 'Cell sets and Metadata',
        direction: 'row',
        splitPercentage: 62,
      },
      direction: 'column',
      splitPercentage: 45,
    },
    second: 'Genes',
    splitPercentage: 70,
  },
  panel: 'Gene list',
};

const initialLayoutSingleCell = {
  windows: {
    direction: 'row',
    first: {
      first: {
        first: 'UMAP', second: 'Cell sets and Metadata', direction: 'row', splitPercentage: 62,
      },
      second: 'Heatmap',
      direction: 'column',
      splitPercentage: 45,
    },
    second: 'Genes',
    splitPercentage: 70,
  },
  panel: 'Gene list',
};

export { initialLayoutSpatial, initialLayoutSingleCell };
