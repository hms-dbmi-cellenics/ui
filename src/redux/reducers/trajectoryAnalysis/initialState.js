const initialState = {
  loading: false,
  error: false,
  plotData: {},
};

export default initialState;
