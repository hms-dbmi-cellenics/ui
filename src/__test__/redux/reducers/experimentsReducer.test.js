import initialState, { experimentTemplate } from 'redux/reducers/experiments/initialState';

import experimentsReducer from 'redux/reducers/experiments';
import { sampleTemplate } from 'redux/reducers/samples/initialState';

import {
  EXPERIMENTS_CREATED,
  EXPERIMENTS_LOADED,
  EXPERIMENTS_LOADING,
  EXPERIMENTS_UPDATED,
  EXPERIMENTS_ERROR,
  EXPERIMENTS_DELETED,
  EXPERIMENTS_SAVING,
  EXPERIMENTS_METADATA_CREATE,
  EXPERIMENTS_METADATA_UPDATE,
  EXPERIMENTS_METADATA_DELETE,
} from 'redux/actionTypes/experiments';

import { SAMPLES_CREATE, SAMPLES_DELETE } from 'redux/actionTypes/samples';

describe('experimentsReducer', () => {
  const experimentId1 = 'experiment-1';
  const experimentId2 = 'experiment-2';

  const experiment1 = {
    id: experimentId1,
    name: 'experiment 1',
    description: 'this is a test description',
    sampleIds: [],
    metadataKeys: [],
    notifyByEmail: true,
    createdAt: '2021-01-01',
    updatedAt: '2022-01-17',
  };

  const experiment2 = {
    id: experimentId2,
    name: 'experiment 2',
    description: 'this is a test description',
    sampleIds: [],
    metadataKeys: [],
    notifyByEmail: true,
    createdAt: '2021-01-01',
    updatedAt: '2022-01-17',
  };

  const sampleId = 'testSampleId';
  const experiment1WithSample = {
    ...experiment1,
    sampleIds: [sampleId],
  };

  const updatedExperiment = {
    ...experiment1,
    name: 'updated name',
    updatedAt: '02-01-2021',
  };

  const oneExperimentState = {
    ...initialState,
    ids: [experimentId1],
    [experimentId1]: experiment1,
    meta: {
      loading: false,
      saving: false,
      error: false,
    },
  };

  const twoExperimentsState = {
    ...initialState,
    ids: [experimentId1, experimentId2],
    [experimentId1]: experiment1,
    [experimentId2]: experiment2,
    meta: {
      loading: false,
      saving: false,
      error: false,
    },
  };

  const oneExperimentWithSampleState = {
    ...initialState,
    ids: [experimentId1],
    [experimentId1]: experiment1WithSample,
    meta: {
      loading: false,
      saving: false,
      error: false,
    },
  };

  const sample = {
    ...sampleTemplate,
    name: 'test sample',
    uuid: sampleId,
    createdAt: '2021-01-01T14:48:00.000Z',
    updatedAt: '2021-01-01T14:48:00.000Z',
  };

  it('Reduces identical state on unknown action', () => expect(
    experimentsReducer(undefined, {
      action: 'well/this/is/not/a/valid/action',
      payload: {},
    }),
  ).toEqual(initialState));

  it('Loads an experiment correctly', () => {
    const newState = experimentsReducer(initialState, {
      type: EXPERIMENTS_LOADED,
      payload: {
        experiments: [experiment1],
      },
    });

    expect(newState.ids).toEqual([experiment1.id]);
    expect(newState[experiment1.id]).toEqual(experiment1);
    expect(newState).toMatchSnapshot();
  });

  it('Overwrites existing state on loading experiments', () => {
    const newState = experimentsReducer(oneExperimentState, {
      type: EXPERIMENTS_LOADED,
      payload: {
        experiments: [experiment2],
      },
    });

    expect(newState.ids).toEqual([experiment2.id]);
    expect(newState[experiment2.id]).toEqual(experiment2);
    expect(newState).toMatchSnapshot();
  });

  it('Loads experiments correctly', () => {
    const newState = experimentsReducer(initialState, {
      type: EXPERIMENTS_LOADED,
      payload: {
        experiments: [
          experiment1, experiment2,
        ],
      },
    });
    expect(newState.ids).toEqual([experiment1.id, experiment2.id]);
    expect(newState.meta.activeExperimentId).toEqual(experiment1.id);
    expect(newState).toMatchSnapshot();
  });

  it('Loads 0 experiments correctly', () => {
    const newState = experimentsReducer(initialState, {
      type: EXPERIMENTS_LOADED,
      payload: {
        experiments: [],
      },
    });
    expect(newState.ids).toEqual([]);
    expect(newState.meta.activeExperimentId).toEqual(undefined);
    expect(newState).toMatchSnapshot();
  });

  it('Loading state changes meta state', () => {
    const newState = experimentsReducer(initialState, {
      type: EXPERIMENTS_LOADING,
    });

    expect(newState.meta.loading).toEqual(true);
    expect(newState).toMatchSnapshot();
  });

  it('Error state inserts error correctly', () => {
    const errorMsg = 'Error message';

    const newState = experimentsReducer(initialState, {
      type: EXPERIMENTS_ERROR,
      payload: { error: errorMsg },
    });

    expect(newState.meta.error).toEqual(errorMsg);
    expect(newState).toMatchSnapshot();
  });

  it('Inserts a new experiment correctly', () => {
    const createdExperimentData = {
      id: experiment1.id,
      name: experiment1.name,
      description: experiment1.description,
      createdAt: experiment1.createdAt,
    };

    const newState = experimentsReducer(initialState, {
      type: EXPERIMENTS_CREATED,
      payload: { experiment: createdExperimentData },
    });

    expect(newState.ids).toEqual([experiment1.id]);
    expect(newState[experiment1.id]).toEqual(
      {
        ...experimentTemplate,
        ...createdExperimentData,
      },
    );
    expect(newState).toMatchSnapshot();
  });

  it('Updates an experiment correctly', () => {
    const newState = experimentsReducer(oneExperimentState, {
      type: EXPERIMENTS_UPDATED,
      payload: {
        experimentId: experiment1.id,
        experiment: updatedExperiment,
      },
    });

    expect(newState.ids).toEqual([experiment1.id]);
    expect(newState[experiment1.id]).toEqual(updatedExperiment);
    expect(newState).toMatchSnapshot();
  });

  it('Deletes an experiment correctly', () => {
    const newState = experimentsReducer(twoExperimentsState, {
      type: EXPERIMENTS_DELETED,
      payload: {
        experimentIds: experiment2.id,
      },
    });

    expect(newState.ids).toEqual([experiment1.id]);
    expect(newState).toEqual(oneExperimentState);
    expect(newState).toMatchSnapshot();
  });

  it('Adds new sampleId when sample is created', () => {
    const newState = experimentsReducer(oneExperimentState, {
      type: SAMPLES_CREATE,
      payload: {
        experimentId: experiment1.id,
        sample,
      },
    });

    expect(newState[experiment1.id].sampleIds).toEqual([sample.uuid]);
    expect(newState).toMatchSnapshot();
  });

  it('Adds new sampleId when sample is created and we already have one sample', () => {
    const anotherSample = {
      ...sampleTemplate,
      name: 'another test sample',
      uuid: 'testAnotherSampleId',
      createdAt: '2021-01-01T14:48:00.000Z',
      updatedAt: '2021-01-01T14:48:00.000Z',
    };

    const newState = experimentsReducer(oneExperimentWithSampleState, {
      type: SAMPLES_CREATE,
      payload: {
        experimentId: experiment1.id,
        sample: anotherSample,
      },
    });

    expect(newState[experiment1.id].sampleIds).toEqual([sampleId, anotherSample.uuid]);
    expect(newState).toMatchSnapshot();
  });

  it('Sets up saving state correctly', () => {
    const newState = experimentsReducer({
      ...oneExperimentWithSampleState,
      meta: {
        ...oneExperimentWithSampleState.meta,
        loading: false,
        saving: false,
        error: true,
      },
    }, {
      type: EXPERIMENTS_SAVING,
    });

    expect(newState.meta.error).toBe(false);
    expect(newState.meta.loading).toBe(false);
    expect(newState.meta.saving).toBe(true);
    expect(newState).toMatchSnapshot();
  });

  it('Deletes samples correctly', () => {
    const newState = experimentsReducer(oneExperimentWithSampleState, {
      type: SAMPLES_DELETE,
      payload: {
        experimentId: experiment1.id,
        sampleIds: [sampleId],
      },
    });

    expect(newState[experiment1.id].sampleIds).toHaveLength(0);
    expect(newState).toMatchSnapshot();
  });

  it('Correctly creates project metadata', () => {
    const newMetadataKey = 'metadata-test';

    const stateWithMetadata = {
      ...oneExperimentWithSampleState,
      [oneExperimentWithSampleState[experiment1.id]]: {
        ...oneExperimentWithSampleState[experiment1.id],
        metadataKeys: [],
      },
    };

    const newState = experimentsReducer(stateWithMetadata, {
      type: EXPERIMENTS_METADATA_CREATE,
      payload: {
        key: newMetadataKey,
        experimentId: experiment1.id,
      },
    });

    expect(newState[experiment1.id].metadataKeys).toEqual([newMetadataKey]);
    expect(newState).toMatchSnapshot();
  });

  it('Correctly updates project metadata', () => {
    const oldMetadataKey = 'metadata-old';
    const newMetadataKey = 'metadata-new';
    const stateWithMetadata = {
      ...oneExperimentWithSampleState,
      [oneExperimentWithSampleState[experiment1.id]]: {
        ...oneExperimentWithSampleState[experiment1.id],
        metadataKeys: [oldMetadataKey],
      },
    };

    const newState = experimentsReducer(stateWithMetadata, {
      type: EXPERIMENTS_METADATA_UPDATE,
      payload: {
        oldKey: oldMetadataKey,
        newKey: newMetadataKey,
        experimentId: experiment1.id,
      },
    });

    expect(newState[experiment1.id].metadataKeys).toEqual([newMetadataKey]);
    expect(newState).toMatchSnapshot();
  });

  it('Correctly deletes project metadata', () => {
    const metadataKey = 'metadata-old';
    const stateWithMetadata = {
      ...oneExperimentWithSampleState,
      [oneExperimentWithSampleState[experiment1.id]]: {
        ...oneExperimentWithSampleState[experiment1.id],
        metadataKeys: [metadataKey],
      },
    };

    const newState = experimentsReducer(stateWithMetadata, {
      type: EXPERIMENTS_METADATA_DELETE,
      payload: {
        key: metadataKey,
        experimentId: experiment1.id,
      },
    });

    expect(newState[experiment1.id].metadataKeys).toEqual([]);
    expect(newState).toMatchSnapshot();
  });
});
