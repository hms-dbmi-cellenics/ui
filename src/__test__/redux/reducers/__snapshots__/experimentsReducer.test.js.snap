// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`experimentsReducer Adds new sampleId when sample is created 1`] = `
Object {
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [
      "testSampleId",
    ],
    "updatedAt": "2022-01-17",
  },
  "ids": Array [
    "experiment-1",
  ],
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Adds new sampleId when sample is created and we already have one sample 1`] = `
Object {
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [
      "testSampleId",
      "testAnotherSampleId",
    ],
    "updatedAt": "2022-01-17",
  },
  "ids": Array [
    "experiment-1",
  ],
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Correctly creates project metadata 1`] = `
Object {
  "[object Object]": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [
      "testSampleId",
    ],
    "updatedAt": "2022-01-17",
  },
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [
      "metadata-test",
    ],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [
      "testSampleId",
    ],
    "updatedAt": "2022-01-17",
  },
  "ids": Array [
    "experiment-1",
  ],
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Correctly deletes project metadata 1`] = `
Object {
  "[object Object]": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [
      "metadata-old",
    ],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [
      "testSampleId",
    ],
    "updatedAt": "2022-01-17",
  },
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [
      "testSampleId",
    ],
    "updatedAt": "2022-01-17",
  },
  "ids": Array [
    "experiment-1",
  ],
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Correctly updates project metadata 1`] = `
Object {
  "[object Object]": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [
      "metadata-old",
    ],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [
      "testSampleId",
    ],
    "updatedAt": "2022-01-17",
  },
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [
      "metadata-new",
    ],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [
      "testSampleId",
    ],
    "updatedAt": "2022-01-17",
  },
  "ids": Array [
    "experiment-1",
  ],
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Deletes an experiment correctly 1`] = `
Object {
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [],
    "updatedAt": "2022-01-17",
  },
  "ids": Array [
    "experiment-1",
  ],
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Deletes samples correctly 1`] = `
Object {
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [],
    "updatedAt": "2022-01-17",
  },
  "ids": Array [
    "experiment-1",
  ],
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Error state inserts error correctly 1`] = `
Object {
  "ids": Array [],
  "meta": Object {
    "activeExperimentId": null,
    "error": "Error message",
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Inserts a new experiment correctly 1`] = `
Object {
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "meta": Object {
      "organism": null,
      "type": "10x",
    },
    "metadataKeys": Array [],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [],
    "updatedAt": null,
  },
  "ids": Array [
    "experiment-1",
  ],
  "meta": Object {
    "activeExperimentId": "experiment-1",
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Loading state changes meta state 1`] = `
Object {
  "ids": Array [],
  "meta": Object {
    "activeExperimentId": null,
    "error": false,
    "loading": true,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Loads 0 experiments correctly 1`] = `
Object {
  "ids": Array [],
  "meta": Object {
    "activeExperimentId": undefined,
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Loads an experiment correctly 1`] = `
Object {
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [],
    "updatedAt": "2022-01-17",
  },
  "ids": Array [
    "experiment-1",
  ],
  "meta": Object {
    "activeExperimentId": "experiment-1",
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Loads experiments correctly 1`] = `
Object {
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [],
    "updatedAt": "2022-01-17",
  },
  "experiment-2": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-2",
    "metadataKeys": Array [],
    "name": "experiment 2",
    "notifyByEmail": true,
    "sampleIds": Array [],
    "updatedAt": "2022-01-17",
  },
  "ids": Array [
    "experiment-1",
    "experiment-2",
  ],
  "meta": Object {
    "activeExperimentId": "experiment-1",
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Overwrites existing state on loading experiments 1`] = `
Object {
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [],
    "updatedAt": "2022-01-17",
  },
  "experiment-2": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-2",
    "metadataKeys": Array [],
    "name": "experiment 2",
    "notifyByEmail": true,
    "sampleIds": Array [],
    "updatedAt": "2022-01-17",
  },
  "ids": Array [
    "experiment-2",
  ],
  "meta": Object {
    "activeExperimentId": "experiment-2",
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`experimentsReducer Sets up saving state correctly 1`] = `
Object {
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [],
    "name": "experiment 1",
    "notifyByEmail": true,
    "sampleIds": Array [
      "testSampleId",
    ],
    "updatedAt": "2022-01-17",
  },
  "ids": Array [
    "experiment-1",
  ],
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": true,
  },
}
`;

exports[`experimentsReducer Updates an experiment correctly 1`] = `
Object {
  "experiment-1": Object {
    "createdAt": "2021-01-01",
    "description": "this is a test description",
    "id": "experiment-1",
    "metadataKeys": Array [],
    "name": "updated name",
    "notifyByEmail": true,
    "sampleIds": Array [],
    "updatedAt": "02-01-2021",
  },
  "ids": Array [
    "experiment-1",
  ],
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;
